lib/Resque.php

```diff
@@ -72,7 +72,7 @@ public static function redis()
 		if (is_callable(self::$redisServer)) {
 			self::$redis = call_user_func(self::$redisServer, self::$redisDatabase);
 		} else {
-			self::$redis = new Redis(self::$redisServer, self::$redisDatabase);
+			self::$redis = new Redis(self::$redisServer, self::$redisDatabase, null, self::$auth);
```

Comment:
This is passing $redisDatabase, which causes a `->select()` in constructor. Line 78 passes auth after the select.

Reply:
What's the value of `$redisServer` that triggers the error for you?

Reply:
So when $redisServer is set to 10.181.53.75 and $redisDatabase is set to 1, the error occurs. If the $redisDatabase is 0, it does not fail, because ->select() is not ran.

Reply:
Why not use `tcp://user:pass@host:port/db` as a value? The `user` part of it is ignored, but that wouldn't be part of `$auth` either so that wouldn't matter